pkg/kubectl/service_basic.go

```diff
@@ -225,12 +226,25 @@ func (s ServiceCommonGeneratorV1) StructuredGenerate() (runtime.Object, error) {
 		})
 	}
 
-	// setup default label and selector
+	// setup default label
 	labels := map[string]string{}
 	labels["app"] = s.Name
-	selector := map[string]string{}
-	selector["app"] = s.Name
-
+	// setup selector according to selector flag, use default if nothing is provided
+	// selector will not be added for service type ExternalName
+	isTypeExternalName := s.Type == api.ServiceTypeExternalName
```

Comment:
I will eliminate variable isTypeExternalName, and use the check directly in if block.